File: src/app/conjugation/forms/non-past-polite-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {HIRAGANA} from "../hiragana";

export class NonPastPoliteNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {
        switch (word.wordType) {

            // Verbs
            case WordType.GodanVerb:
                const lastKana = HIRAGANA[word.getLastKana()]
                return word.replaceLastKana(lastKana.getGroup().i + 'ません')

            case WordType.IchidanVerb:
                return word.replaceLastKana('ません');

            case WordType.SuruVerb:
                return word.replaceLastKana('しません', 2);

            case WordType.KuruVerb:
                return word.replace('来ません', 'きません');



            // Adjectives
            case WordType.IAdjective:
                return word.replaceLastKana('くないです');

            case WordType.NaAdjective:
                return word.addSuffix('じゃないです');


            default:
                return undefined;
        }
    }

    getTitle(): string {
        return "Non-past, polite, negative"
    }

    getSettingsTitle(): string {
        return "Polite, negative"
    }

}

File: backend/app/conjugation/forms/non_past_polite_negative.py
from ..core import Conjugation, Word, WordType
from ..hiragana import HIRAGANA


class NonPastPoliteNegative(Conjugation):

    title = 'Non-past, polite, negative'
    settings_title = 'Polite, negative'

    def conjugate(self, word: Word) -> Word | None:
        match word.word_type:

            # Verbs
            case WordType.GODAN_VERB:
                last_kana = HIRAGANA[word.last_kana]
                return word.replace_last_kana(last_kana.group.i + 'ません')

            case WordType.ICHIDAN_VERB:
                return word.replace_last_kana('ません')

            case WordType.SURU_VERB:
                return word.replace_last_kana('しません', 2)

            case WordType.KURU_VERB:
                return word.replace('来ません', 'きません')

            # Adjectives
            case WordType.I_ADJECTIVE:
                return word.replace_last_kana('くないです')

            case WordType.NA_ADJECTIVE:
                return word.add_suffix('じゃないです')

            case _:
                return None
